syntax = "proto3";
option optimize_for = CODE_SIZE;

package protorpc;

import "google/protobuf/empty.proto";
import "google/protobuf/any.proto";

enum RpcErrorType {
    SERVER_TIMEOUT = 0;
    CLIENT_TIMEOUT = 1;
    UNKNOWN_RPC_METHOD = 2;
    CLIENT_ERROR = 3;
    SERVER_ERROR = 4;
  }

message RpcMessage {
    map<string, string> header = 1;
    google.protobuf.Any body = 2;
    string requestId = 3;
    optional RpcErrorType errorType = 4;
    optional string errorClassName = 5;
    optional string errorCode = 6;
    optional string errorMessage = 7;
  }

// Dummy message to force the generation of the typescript class "google.prototype.Empty"
// We need this generated class in RpcCommunicator

message Mnfo2uhnf92hvqi2nviouq2hv9puhq {
  google.protobuf.Empty empty = 1;
}